extensions/amp-story/1.0/background-blur.js

```diff
@@ -84,22 +90,39 @@ export class BackgroundBlur {
    * @param {!Element} pageElement
    */
   update(pageElement) {
-    const ampImgEl = this.getBiggestImage_(pageElement);
-    if (!ampImgEl) {
-      user().info(CLASS_NAME, 'No image found for background blur.');
+    const mediaEl = this.getBiggestMediaEl_(pageElement);
+    if (!mediaEl) {
+      user().info(CLASS_NAME, 'No image or video found for background blur.');
       this.animate_();
       return;
     }
 
-    // Ensures img element exists and is loaded.
-    whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    // Ensure element is loaded before calling animate.
+    whenUpgradedToCustomElement(mediaEl)
+      .then(() => mediaEl.signals().whenSignal(CommonSignals.LOAD_END))
       .then(
         () => {
-          this.animate_(ampImgEl.querySelector('img'));
+          if (mediaEl.tagName === 'AMP-IMG') {
+            // If image render right away.
+            this.animate_(mediaEl.querySelector('img'));
+          } else {
+            const innerVideoEl = mediaEl.querySelector('video');
+            const alreadyHasData = innerVideoEl.readyState >= HAVE_CURRENT_DATA;
+            if (alreadyHasData) {
+              // If video already has data, render first frame.
+              this.animate_(innerVideoEl);
+            } else {
+              // If video doesnt have data, render from the poster image.
+              const img = new Image();
+              img.src = mediaEl.getAttribute('poster');
```

Comment:
Plz set the src after you register the callback, it's prob working perfect cross browser now but I remember having issues with this in the past, and still a couple of years ago with something similar on iframes